src/matlab2tikz.m

```diff
@@ -407,9 +407,19 @@ function matlab2tikz(varargin)
 end
 % ==============================================================================
 function fid = fileOpenForWrite(m2t, filename)
-    encoding = switchMatOct({'native', m2t.cmdOpts.Results.encoding}, {});
```

Comment:
Maybe `cell(0)` instead of `{}` does the trick? Quite strange that Octave 4 no longer supports that syntax for an empty cell. I don't think that `[]` would work, though.